### PR TITLE
Add warning when overriding page path

### PR DESCRIPTION
**Pull request by tsvetlin**
## Description
There was an already existing pull request for it: [https://github.com/gatsbyjs/gatsby/pull/8897](https://github.com/gatsbyjs/gatsby/pull/8897)

It reuses the same logic, but also updates the documentation.

## Related Issues
Closes: #5836

### CODE REVIEW

File: docs/tutorial/part-three/index.md [hunk 1]
```diff
@@ -232,7 +232,7 @@ const ListLink = props => (
 // highlight-end
 
 export default ({ children }) => (
-  <div style={{ margin: `0 auto`, maxWidth: 650, padding: `1.25rem 1rem` }}>
+  <div style={{ margin: `3rem auto`, maxWidth: 650, padding: `0 1rem` }}>
```
Comment (pieh) [inline docs/tutorial/part-three/index.md]: Ops, I thought we branched out of master, apparently we did something wrong then, but let's remove this change from this PR

Comment (tsvetlin) [reply docs/tutorial/part-three/index.md]: I missed to switch the branch. Sorry. I'll fix that.

Comment (tsvetlin) [reply docs/tutorial/part-three/index.md]: Updated the pull request.

**Comment by pieh** [comment]
We should also `deletePage` in this plugin https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-create-client-paths/src/gatsby-node.js#L32 (as otherwise it would trigger this new warning message)

**Comment by tsvetlin** [comment]
> We should also `deletePage` in this plugin https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-create-client-paths/src/gatsby-node.js#L32 (as otherwise it would trigger this new warning message)

I'll fix this too. Thank you for pointing it out!

**Comment by tsvetlin** [comment]
I fixed this too.

**Comment by pieh** [comment]
I was thinking a bit about this and I think we would show a lot of warnings in `gatsby develop` (when we rerun `createPages` API hook). I don't have good solution yet, will need to think more about it

**Comment by LekoArts** [comment]
Hey @pieh 👋 	Any updates on that? :)

**Comment by sidharthachatterjee** [comment]
Thank you so much for opening this pull request @tsvetlin 👍 

This is definitely a valid issue (I've reopened the linked issue for now) but this change will result in too many false positives (most plugins update pages at the moment with multiple calls to `createPage` and _do not_ call `deletePage` before they do)

We think the false positives outweigh the benefit of this warning so let's close this till we figure out how to best handle this! If you like, we can continue this discussion on the issue at https://github.com/gatsbyjs/gatsby/issues/5836

Thank you so much nonetheless for opening this! We value your contributions and would love to see you around 💜 💪

**Pull request closed without merging**